near-sdk/src/store/lazy_option/impls.rs

```diff
@@ -30,3 +30,13 @@ where
         Self::get_mut(self)
     }
 }
+
+#[cfg(feature = "expensive-debug")]
+impl<T> std::fmt::Debug for LazyOption<T>
+where
+    T: std::fmt::Debug + BorshSerialize + BorshDeserialize,
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.get().fmt(f)
+    }
+}
```

Comment:
```suggestion
impl<T> std::fmt::Debug for LazyOption<T>
where
    T: std::fmt::Debug + BorshSerialize + BorshDeserialize,
{
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        if cfg!(feature = "expensive-debug") {
            self.get().fmt(f)
        } else {
            // Format just the metadata
        }
    }
}
```

Maybe there is a misconception, this feature should be used to distinguish between an expensive but more understandable debug output. For example, `Vector` expensive debug will load all values, allocate them in a vector, and then print that when otherwise it would just print metadata. Debug impls shouldn't be expensive by default, like in this, but expensive debug can incur storage costs for debugging reasons to see the data in testing.

Reply:
Thanks for the clarification. Indeed, I didn't realize that we want both expensive and non-expensive Debug implementations.